src/ui/context_menu/warehouse_spec_menu.gd

```diff
@@ -0,0 +1,27 @@
+extends MarginContainer
+
+var selected_warehouse: WarehouseBuilding = null
+var main_panel: BuildingInfoPanel = null
+
+func _ready() -> void:
+	main_panel = get_parent().get_parent().get_parent()
+
+func _on_water_button_pressed() -> void:
+	SoundManager.play_button_click_sfx()
```

Comment:
We could add a generic `_on_resource_button_pressed()` method that takes a resource type as an input and call it from each of these methods. Just to reduce code repetition.